MIP/mip-58/README.md

```diff
@@ -0,0 +1,304 @@
+   - Removes the refund functionality entirely to eliminate associated exploits.
+
+3. **Parameter validation**
+   - Ensure parameter validation on the counterparty to prevent invalid transactions.
+
+4. **Relayer redundancy**
+   - Use two types of Relayers:
+     - **Automated relayer**: Operated with minimal human involvement; its private key is highly secured.
+     - **Multi-signature Relayer**: Managed by the team to guarantee transaction completion in case of failures.
+
+5. **Cost efficiency**
+   - Minimize gas costs by reducing the number of interactions and simplifying fee calculations.
+
+6. **Enhanced security**
+   - Avoid refund logic to close exploit windows.
+   - Protect against key compromise through key isolation or known by no parties and multi-signature Relayer setups.
+   - There is no scenario where a bridge could lead to double-spending. It's either completed by Relayer or not.
+   - In the HTLC bridge the user can loose its bridge `preImage` which could lead to them being unable to complete the bridge. By not relying on a `preImage` from the user, it minimizes issues. It is not a loss in security because the purpose of the `preImage` is solely for refunding.
+
+7. **Batch processing for downtime**
+   - Multi-signature Relayers can process multiple pending transactions in a single batch to compensate for downtime.
+
+8. **BridgeTransferId**
+   - Continue using unique identifier to prevent double-spending and track transactions securely.
+
+9. **Initial bridge fee proposals**
+   - On the L1 to L2 bridge, do not charge fees.
+   - On the L2 to L1 bridge, charge a fee estimated by admin. It's set to the gas spent in ethereum in move. This requires an oracle and can only be implemented after oracles are live and we are able to have a maintainer that is able to set the fees on L2.
+
+   ```
+   entry fun set_fee(caller: &signer, fee: u64) {
+      assert_is_maintainer(caller);
+      borrow_global<BridgeConfig>(@aptos_framework).fee = fee;
+   }
+   ```
+
+10. **Best practices**
+
+- Adopt currently used bridge designs from established designs like Arbitrum, LayerZero and Blast bridges, which use a Relayer to finalize the bridge.
+- User is not required to have funds on counterparty contract to finalize the bridge.
+
+## Exploits and potential losses
+
+1. **Key compromise**
+   - The compromise of the Relayer keys would lead to unauthorized transactions, putting the protocol at risk of unlimited value exploit.  The protocol must absorb the losses and rotate Relayers. To minimize this risk [MIP-74](https://github.com/movementlabsxyz/MIP/pull/74) proposes a rate limitation even for the Relayer.
+2. **Fee wrong estimation**
+   - Incorrect fee calculations (e.g., underestimating gas) can cause significant financial losses for the operator.
+
+## Reference implementation
+
+[Solidity Implementation](https://github.com/movementlabsxyz/movement/tree/primata/simple-native-bridge)
```

Comment:
as per discussion with @andygolay we should reference specific commits. Could you recommend a specific commit? 

also for the Move Implementation